File: JavaOOP_modificators/app/src/main/java/com/example/modificators/MainActivity.java
package com.example.modificators;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Cat cat1 = new Cat();
        Log.i("numberOfLegs", "number of legs cat1 is: " + cat1.numberOfLegs); // 4
        cat1.numberOfLegs--; // но так обращатся не правильно, лучше делать так  Cat.numberOfLegs--. Тогда будет радобать автокомплит
        Log.i("numberOfLegs", "number of legs cat1 is: " + cat1.numberOfLegs); // 3

        Cat cat2 = new Cat();
        Log.i("numberOfLegs", "number of legs cat2 is: " + cat2.numberOfLegs); // 3

        Log.i("countOfInstance", String.valueOf(Cat.countOfInstance)); // 2

        Cat.canBreath = false; // Мы не может менять значение этого свойства

        Lion lion = new Lion();
        lion.talk();
    }
}

File: Kotlin new/ErrorBoundary/app/src/main/java/com/example/errorboundary/MainActivity.kt
package com.example.errorboundary


import android.content.Intent
import android.graphics.PixelFormat
import android.os.Build
import android.os.Bundle
import android.os.Process
import android.view.Gravity
import android.view.WindowManager
import android.widget.Button
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import kotlin.system.exitProcess


class MainActivity : AppCompatActivity() {

    @RequiresApi(Build.VERSION_CODES.N)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


//        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
//            // set the layout parameters of the window
//            val mParams = WindowManager.LayoutParams( // Shrink the window to wrap the content rather
//                // than filling the screen
//                WindowManager.LayoutParams.WRAP_CONTENT,
//                WindowManager.LayoutParams.WRAP_CONTENT,  // Display it on top of other application windows
//                WindowManager.LayoutParams.TYPE_APPLICATION_OVERLAY,  // Don't let it grab the input focus
//                WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE,  // Make the underlying application window visible
//                // through any transparent parts
//                PixelFormat.TRANSLUCENT
//            )
//
//
//            mParams.gravity = Gravity.CENTER
//            this.window.attributes = mParams
//        }



        findViewById<Button>(R.id.make_error_button).setOnClickListener {
            makeError()
        }


        Thread.setDefaultUncaughtExceptionHandler { paramThread, paramThrowable ->
            val intent = Intent(this, ErrorActivity::class.java)
            startActivity(intent)
            killCurrentProcess() // work on my phone (if use only exitProcess I get xiomi bottomsheet)
        }


    }

    private fun killCurrentProcess() {
        Process.killProcess(Process.myPid())
        // any number
        exitProcess(0)
    }

    private fun makeError() {
        throw Error("My Error")
    }

}